### DISCUSSION TITLE
Q: remove php-imap requirement

### DISCUSSION THREAD

**Discussion by vladsf** [eventum/eventum#936] [Ideas]
Hi,

php-imap package removed from RHEL8 and derivatives.
Getting PHP 7.2 with php-imap extension is not out of box anymore.
php-imap depends on a pretty old libc-client which in turn requires openssl 1.0 which makes things troublesome in the near future.
https://bugzilla.redhat.com/show_bug.cgi?id=1755408

Is there any plan to workaround obsoleting php-imap and libc-client?

**Comment by glensc** [comment]
no actual plan. switching to zend\mail was so overwhelming that switching pop3/imap part was just forgotten.

i think pieces are even in place to use zend\mail for this.

**Comment by glensc** [comment]
some links:
- https://github.com/glpi-project/glpi/issues/3481
- https://github.com/glpi-project/glpi/pull/6118

**Comment by glensc** [comment]
actually the alternative planned to use zend (now laminas) components, but they plan to put the mail component to security-only:
- https://github.com/laminas/technical-steering-committee/pull/31

and so far I have not found any alternative to parse input emails.

we migrated from pear/mail, which was huge project, took several iterations to complete (3.4.0 2018-04-22):
- Drop `pear/mime_decode`, `pear/mail_mime` in favor of `Zend\Mail` (@glensc, #263, #333, #334, #335)

**Comment by vladsf** [comment]
I guess moving from an old php-imap used by many  to obsoleting laminas component is not a good idea. php-imap [gets patches](https://github.com/uw-imap/imap/tree/cab109466534e206a3652ef1c68fe88101b68bda) and many face the same question.
I can see it gets openssl 1.1 patch and thus it is not abandoned and I could be wrong in my initial post.
